registry/reference.go

```diff
@@ -98,56 +98,92 @@ func ParseReference(artifact string) (Reference, error) {
 	}
 	registry, path := parts[0], parts[1]
 
+	var isTag bool
 	var repository string
 	var reference string
 	if index := strings.Index(path, "@"); index != -1 {
 		// `digest` found; Valid Form A (if not B)
+		isTag = false
 		repository = path[:index]
 		reference = path[index+1:]
 
-		if index := strings.Index(repository, ":"); index != -1 {
+		if index = strings.Index(repository, ":"); index != -1 {
 			// `tag` found (and now dropped without validation) since `the
 			// `digest` already present; Valid Form B
 			repository = repository[:index]
 		}
-	} else if index := strings.Index(path, ":"); index != -1 {
+	} else if index = strings.Index(path, ":"); index != -1 {
 		// `tag` found; Valid Form C
+		isTag = true
 		repository = path[:index]
 		reference = path[index+1:]
 	} else {
 		// empty `reference`; Valid Form D
 		repository = path
 	}
-	res := Reference{
+	var ref = Reference{
 		Registry:   registry,
 		Repository: repository,
 		Reference:  reference,
 	}
-	if err := res.Validate(); err != nil {
+
+	var err error
+
+	if err = ref.ValidateRegistry(); err != nil {
```

Comment:
It is better to have
```suggestion
	if err := ref.ValidateRegistry(); err != nil {
```

Reply:
Fixed.